src/main/drivers/pwm_output.c

```diff
@@ -280,3 +281,28 @@ void servoInit(const servoConfig_t *servoConfig)
 }
 
 #endif
+
+#ifdef BRUSHED_ESC_AUTODETECT
+uint8_t hardwareMotorType = MOTOR_UNKNOWN;
```

Comment:
@AndersHoglund : Using smaller types is commin in CF, mainly as optimization. Using `__attribute__ ((packed))` enums will get identical memory savings and type correctness. And hiding the attribute in macro will allow the code to work on non-gcc compiler.
It may bring unexpected problems with type size, but it is probably much better than overflowing `uint8_t` range with enum (quite easy with bitmask enums)